tests/benchmark/generic/generic_test.go

```diff
@@ -0,0 +1,13 @@
+package generic
+
+import (
+	"testing"
+
+	"github.com/thefuga/go-collections"
+)
+
+func BenchmarkRange(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		_ = collections.Range[int](-i, i)
```

Comment:
I think this might be optimized by the compiler and end up not being called as expected. It's not totally predictable but I've had issues with this in some benchmarks.
Take a look [here](https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go)

Reply:
Thanks, I understand the reason and will make the changes.